### PR TITLE
Pre demo

### PR DESCRIPTION
**Pull request by jack-y-xu**
Fixed Leaderboard returning NaN, Fixed not displaying end-game screen early enough in Host page, Added button styling for button in Host page that was missed earlier. Added automatic +1 round to the create page so game logic works for the last round.

### CODE REVIEW

File: src/components/firebase-access.jsx [hunk 1]
```diff
@@ -250,8 +250,9 @@ export const setUpRoom = (numOfSymbols,rounds,password,startingMoney = 10000) =>
 
 //TODO: this method can be called many times, which leads to latency due to a lot of awaits. See if can pass userData from other methods
 export const updateNetWorth = async (roomID, userID) => {
+    const numDays = await getNumDays(roomID);
     const dayIndex = await getDayIndex(roomID);
-    const prices = await getPrices(roomID, dayIndex);
+    const prices = await getPrices(roomID, dayIndex>= numDays ? numDays-1 : dayIndex);
```
Comment (asam01) [inline src/components/firebase-access.jsx]: nit
```suggestion
    const prices = await getPrices(roomID, dayIndex >= numDays ? numDays - 1 : dayIndex);
```

File: src/components/firebase-access.jsx [hunk 2]
```diff
@@ -44,7 +44,7 @@ export const advanceDay = async (roomID) => {
     });
 
     var roomData = await getRoomData(roomID);
-    if (roomData.dates.length == roomData.day_index) {
+    if (roomData.dates.length - 1<= roomData.day_index) {
```
Comment (asam01) [inline src/components/firebase-access.jsx]:
```suggestion
    if (roomData.dates.length - 1 <= roomData.day_index) {
```

File: src/components/firebase-access.jsx [hunk 3]
```diff
@@ -250,8 +250,9 @@ export const setUpRoom = (numOfSymbols,rounds,password,startingMoney = 10000) =>
 
 //TODO: this method can be called many times, which leads to latency due to a lot of awaits. See if can pass userData from other methods
 export const updateNetWorth = async (roomID, userID) => {
+    const numDays = await getNumDays(roomID);
     const dayIndex = await getDayIndex(roomID);
-    const prices = await getPrices(roomID, dayIndex);
+    const prices = await getPrices(roomID, dayIndex>= numDays ? numDays - 1 : dayIndex);
```
Comment (asam01) [inline src/components/firebase-access.jsx]:
```suggestion
    const prices = await getPrices(roomID, dayIndex >= numDays ? numDays - 1 : dayIndex);
```

File: src/components/pages/Host.jsx [hunk 4]
```diff
@@ -168,14 +168,18 @@ class Host extends Component {
...
+        const {roomId,questionNum,numDays} = this.state;
         const that = this;
         this.setState({
             questionNum: that.state.questionNum + 1,
         });
-        if (questionNum >= (await getNumDays(roomId)))
+        console.log("questionNum");
+        console.log(questionNum);
+        console.log("numDays");
+        console.log(numDays);
```
Comment (asam01) [inline src/components/pages/Host.jsx]:
```suggestion
```

File: src/components/pages/Host.jsx [hunk 5]
```diff
@@ -168,14 +168,18 @@ class Host extends Component {
...
+        if (questionNum+2 >= numDays)
```
Comment (asam01) [inline src/components/pages/Host.jsx]:
```suggestion
        if (questionNum + 2 >= numDays)
```

File: src/components/pages/Host.jsx [hunk 6]
```diff
@@ -168,14 +168,18 @@ class Host extends Component {
...
             this.setState({
-                phase: "ended",
+                phase: 'ended',
             })
```
Comment (asam01) [inline src/components/pages/Host.jsx]: semi colon consistency
```suggestion
            });
```

File: src/components/pages/Host.jsx [hunk 7]
```diff
@@ -240,13 +244,13 @@ class Host extends Component {
                     <div style={styles.box} className="page-container host-page">
                         <span>Current question: </span>
                         {' '}
-                        <span className="dynamic-text">{questionNum}</span>
+                        <span className="dynamic-text">{questionNum+1}</span>
```
Comment (asam01) [inline src/components/pages/Host.jsx]:
```suggestion
                        <span className="dynamic-text">{questionNum + 1}</span>
```

File: src/components/pages/Play.jsx [hunk 8]
```diff
@@ -141,7 +141,7 @@ class Play extends Component {
                         <div style={styles.box}>
                             <FormControl style={{margin:'1rem 1rem 1rem 1rem'}}>
                                 <TextField label="Nickname" name="nickname" value={nickname}
-                                           onChange={this.handleChange('roomId')}/>
+                                           onChange={this.handleChange('nickname')}/>
```
Comment (asam01) [inline src/components/pages/Play.jsx:144]: lol nice fix

File: src/components/quiz/create/Config.jsx [hunk 9]
```diff
@@ -28,7 +28,7 @@ class Config extends React.Component {
     createRoom = () => {
         console.log("create room called");
         const {numSymbols,numRounds} = this.state;
-        var roomID = setUpRoom(numSymbols,numRounds,'');
+        var roomID = setUpRoom(numSymbols,numRounds+1,'');
```
Comment (asam01) [inline src/components/quiz/create/Config.jsx]:
```suggestion
        var roomID = setUpRoom(numSymbols,numRounds + 1,'');
```

File: src/components/quiz/create/Config.jsx [hunk 10]
```diff
@@ -28,7 +28,7 @@ class Config extends React.Component {
...
         this.props.updatePageType('created',roomID);
         console.log(this.props.roomId);
```
Comment (asam01) [inline src/components/quiz/create/Config.jsx]:
```suggestion
```

**Review by asam01 — COMMENTED** [review]
couple of nits

File: src/components/pages/Play.jsx (hunk 8 above)
Comment (jack-y-xu) [reply src/components/pages/Play.jsx:144]: ¯\\_(ツ)_/¯

**Review by asam01 — APPROVED** [review]

**Review by yukyhaha — APPROVED** [review]

**Pull request merged**